DSCResources/MSFT_xDefaultGatewayAddress/MSFT_xDefaultGatewayAddress.psm1

```diff
@@ -187,77 +212,92 @@ function Test-Properties
             # Is a default gateway address set?
             if ($defaultRoutes)
             {
-                Write-Verbose -Message "Default gateway exists but it should not."
+                Write-Verbose -Message ( @("$($MyInvocation.MyCommand): "
+                    'Default gateway exists but it should not.'
+                    ) -join '' )
                 $requiresChanges = $true
             }
             else
             {
-                Write-Verbose -Message "Default Gateway does not exist which is correct."
+                Write-Verbose -Message ( @("$($MyInvocation.MyCommand): "
+                    'Default Gateway does not exist which is correct.'
+                    ) -join '' )
             }
         }
+    }
+    catch
+    {
+        Write-Verbose -Message ( @("$($MyInvocation.MyCommand): "
+            'Error testing valid Default Gateway address using InterfaceAlias $InterfaceAlias '
+            "and AddressFamily $AddressFamily"
+            ) -join '' )
+        throw $_.Exception
+    }
+    return -not $requiresChanges
+}
 
-        if($requiresChanges)
-        {
-            # Apply is true in the case of set - target resource - in which case, it will apply the
-            # required IP configuration
-            if($Apply)
-            {
-                Write-Verbose -Message ( @(
-                    "At least one setting differs from the passed parameters. Applying "
-                    "configuration..."
-                    ) -join ""
-                )
+#######################################################################################
+#  Helper functions
+#######################################################################################
+function Test-ResourceProperty {
```

Comment:
:+1:  All your previous comments I've applied to this resource (as well as xDefaultGatewayAddress) :)